base/daemon/SentryWriter.cpp

```diff
@@ -0,0 +1,244 @@
+        sentry_value_t event = sentry_value_new_message_event(SENTRY_LEVEL_FATAL, "fault", error_message.c_str());
+        sentry_set_tag("signal", strsignal(sig));
+        sentry_set_extra("signal_number", sentry_value_new_int32(sig));
+        setExtras();
+
+        /// Prepare data for https://develop.sentry.dev/sdk/event-payloads/stacktrace/
+        sentry_value_t sentry_frames = sentry_value_new_list();
+        size_t stack_size = stack_trace.getSize();
+        if (stack_size > 0)
+        {
+            size_t offset = stack_trace.getOffset();
+            if (stack_size == 1)
+            {
+                offset = 1;
+            }
+            char instruction_addr[100];
+            StackTrace::Frames frames;
+            StackTrace::symbolize(stack_trace.getFramePointers().data(), offset, stack_size, frames);
+            for (size_t i = stack_size - 1; i >= offset; --i)
+            {
+                const StackTrace::Frame & current_frame = frames[i];
+                sentry_value_t sentry_frame = sentry_value_new_object();
+                UInt64 frame_ptr = reinterpret_cast<UInt64>(current_frame.virtual_addr);
+                std::snprintf(instruction_addr, sizeof(instruction_addr), "0x%" PRIx64, frame_ptr);
+                sentry_value_set_by_key(sentry_frame, "instruction_addr", sentry_value_new_string(instruction_addr));
+
+                if (current_frame.symbol.has_value())
+                {
+                    sentry_value_set_by_key(sentry_frame, "function", sentry_value_new_string(current_frame.symbol.value().c_str()));
+                }
+
+                if (current_frame.file.has_value())
+                {
+                    sentry_value_set_by_key(sentry_frame, "filename", sentry_value_new_string(current_frame.file.value().c_str()));
+                }
+
+                if (current_frame.line.has_value())
+                {
+                    sentry_value_set_by_key(sentry_frame, "lineno", sentry_value_new_int32(current_frame.line.value()));
+                }
+
+                sentry_value_append(sentry_frames, sentry_frame);
+            }
+        }
+
+        /// Prepare data for https://develop.sentry.dev/sdk/event-payloads/threads/
+        sentry_value_t stacktrace = sentry_value_new_object();
+        sentry_value_set_by_key(stacktrace, "frames", sentry_frames);
+
+        sentry_value_t thread = sentry_value_new_object();
+        sentry_value_set_by_key(thread, "stacktrace", stacktrace);
+
+        sentry_value_t values = sentry_value_new_list();
+        sentry_value_append(values, thread);
+
+        sentry_value_t threads = sentry_value_new_object();
+        sentry_value_set_by_key(threads, "values", values);
+
+        sentry_value_set_by_key(event, "threads", threads);
+
+        LOG_INFO(logger, "Sending crash report");
+        sentry_capture_event(event);
+        shutdown();
+    }
+    else
+    {
+        LOG_INFO(logger, "Not sending crash report");
```

Comment:
It's better to state that we are sending crash reports to Sentry in all log messages.
Otherwise it is not clear enough for user.

Reply:
@alexey-milovidov Sentry is mentioned in logger name, why repeating it will make the situation more clear?

Reply:
Ok.